### DISCUSSION TITLE
Feature Suggestion - Virtual Scrolling for speed & 1 million nodes

### DISCUSSION THREAD

**Discussion by jplevene** [vakata/jstree#2587] [Ideas]
**IGNORE & DELETE THIS AS I SAW THIS WILL BE HAPPENING IN VERSION 4**

Saw another plugin that did this, and was really impressed.  It's sort of a bit like REACT

Basically, only nodes that are visible would be rendered in the DOM, everything above and below the visible pane is just a single div element, set to the out of view nodes height.

This would enable jsTree to have 1 million open nodes and work really fast, just like there were 20.

This is a demo of a table plugin that does this: [https://paramquery.com/pro/demos/infiniteV](https://paramquery.com/pro/demos/infiniteV).  

Above are 1 million rows and it renders and scrolls like there are 6, because in essence, there are only 6 rows + the space above and below in the DOM.

I haven't looked too hard into the example, but it seems to remove a row from the DOM when it goes out view and create one at the other end in the DOM when it comes into view.

**Comment by vakata** [comment]
I just came by to tell you to check out v4 - it already has this functionality.

**Reply by jplevene** [reply]
> I just came by to tell you to check out v4 - it already has this functionality.

Thanks. Does V4 have the plugins working yet?  I can see that it hasn't been touched for 8 months.
